### PR TITLE
connectedhomeip: samples: Added light bulb and light switch samples.

### PR DESCRIPTION
**Pull request by kkasperczyk-no**
* Added light bulb sample
* Added light switch sample
* Added support for nRF5340
* Updated CHIP revision
* Fixed a couple of issues in the CHIP samples:
  * switched to Oberon SHA256 to workaround a hardware limitation (CC310 unable to read flash memory)
  * erasing Thread configuration on init not to leave a device in a partially persisted state (CHIP doesn't support persistent storage yet)
  * removed spaces from NFC URI Tag

To reviewers: files in the "gen/" directories are auto-generated using the [ZAP](https://github.com/project-chip/zap) tool, so detailed review of their contents probably doesn't make much sense.

### CODE REVIEW

**Comment by kkasperczyk-no** [comment]
Do not remove DNM label until sdk-connectedhomeip version will be updated.

**Comment by doublemis1** [comment]
sample.yamls looks fine

File: samples/connectedhomeip/light_bulb/CMakeLists.txt
Comment (LuDuda) [inline samples/connectedhomeip/light_bulb/CMakeLists.txt:18]: This can be removed now.

File: samples/connectedhomeip/light_bulb/README.rst
Comment (LuDuda) [inline samples/connectedhomeip/light_bulb/README.rst:114]: Should we mention somewhere, what is the test mode? I.e. that Commissioning process is bypassed, and keys hardcoded.

File: samples/connectedhomeip/light_bulb/src/chip_project_config.h
Comment (LuDuda) [inline samples/connectedhomeip/light_bulb/src/chip_project_config.h:23]: nit: alignment

File: samples/connectedhomeip/light_bulb/src/main.cpp
Comment (LuDuda) [inline samples/connectedhomeip/light_bulb/src/main.cpp:45]: Can we add it to the docuemntation as well for both examples?

File: samples/connectedhomeip/light_bulb/src/light_bulb_publish_service.cpp
Comment (LuDuda) [inline samples/connectedhomeip/light_bulb/src/light_bulb_publish_service.cpp:22]: Thi should be changed to `ff03::1`. After that, we should also update the doc, changing `fe80..` to `fd..`.

File: samples/connectedhomeip/light_bulb/README.rst
Comment (greg-fer) [reply samples/connectedhomeip/light_bulb/README.rst]: I think this is already covered by the reused lines from the door lock sample:

Remote testing using test mode
==============================

.. include:: ../lock/README.rst
    :start-after: chip_door_lock_sample_test_mode_start
    :end-before: chip_door_lock_sample_test_mode_end

Which expands to:

Alternatively to the commissioning procedure, you can use the test mode, which allows to join a Thread network with default static parameters and static cryptographic keys. Use Button 3 to enable this mode after building and running the sample.

**Review by greg-fer — APPROVED** [review]

**Review by LuDuda — APPROVED** [review]

**Review by doublemis1 — APPROVED** [review]

**Comment by rlubos** [comment]
@LuDuda Why did you remove DNM label? The manifest file still  points to a pull request.

**Review by rlubos — COMMENTED** [review]
LGTM, pretty neat. I can approve after the manifest is updated with proper SHA.

**Comment by Damian-Nordic** [comment]
@rlubos I updated the manifest with the final SHA. We were not sure if we should first wait for review and then merge changes to sdk-connectedhomeip or the other way around.

**Review by rlubos — APPROVED** [review]

**Pull request merged**
